connect_timeout_test.go

```
@@ -0,0 +1,159 @@
+// network; the dialer-bounds-connect behavior itself is proven by the QWP
+// black-hole tests, which share the same net.Dialer{Timeout} mechanism.)
+func TestConnectTimeoutHttpUsesPrivateTransport(t *testing.T) {
+	c, err := confFromStr("http::addr=192.0.2.1:9009;protocol_version=2;connect_timeout=500;")
+	if err != nil {
+		t.Fatalf("conf: %v", err)
+	}
+	ls, err := newHttpLineSender(context.Background(), c)
+	if err != nil {
+		t.Fatalf("build: %v", err)
+	}
+	defer ls.Close(context.Background())
```

golangci-lint (2.12.2)

[error] Error return value of `ls.Close` is not checked

(errcheck)